data/serializer/src/main/java/tech/pegasys/teku/api/schema/ValidatorsRequest.java

```diff
@@ -36,5 +37,10 @@ public ValidatorsRequest(
       @JsonProperty(value = "pubkeys", required = true) final List<BLSPubKey> pubkeys) {
     this.epoch = epoch;
     this.pubkeys = pubkeys;
+    // Restrict valid epoch values to ones that can be converted to slot without overflowing
+    if (epoch != null
+        && epoch.isGreaterThan(UInt64.MAX_VALUE.dividedBy(Constants.SLOTS_PER_EPOCH))) {
+      throw new IllegalArgumentException("Epoch is too large.");
+    }
```

Comment:
the endpoint creating this request should probably be calling the above query util to get the epoch argument? then this check is probably not required as its already validated...

Reply:
actually, it must be json parsing input, never mind, we'll need to do this here.

Reply:
Yep, and I couldn't even reuse the code because it was in different modules. :(